curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/660f487dc0c8fa38084f9754.md

```diff
@@ -14,7 +14,7 @@ Since you have moved the comparison into the `while` condition, you can remove y
 You should no longer have an `if` statement.
 
 ```js
-assert.notMatch(__helpers.removeJSComments(code), /if\s*\(\s*done\)/);
+assert.notMatch(__helpers.removeJSComments(code), /if\s*\(\s*done\s*===\s*count\s*\)/);
```

Comment:
I'm fine with it how it is, but I don't particularly like this regex. The task is to remove the entire `if` - if they just change a tiny bit of it, it passes - e.g. change `count` to `coun`. We don't want them to have `if`, `done`, `count` or any of it. I think I would test this by checking the surrounding code - make sure they have something like this:

```
  rows.push(padRow(done, count));
}
let result = ""
```

But maybe that's too restrictive. I'll let you fix it if you want, @gikf - or like I said, it's pretty minor so I'm fine with it.

Reply:
That's really good point. I tried to write regex, which would allow optionally the `continueLoop = false;` from the inside `if`, but it's a bit spotty once parts of it are getting removed:
```suggestion
assert.match(__helpers.removeJSComments(code), /rows\.push\s*\(padRow\s*\(\s*done\s*,\s*count\s*\)\s*\)\s*;?\s*(?:continueLoop\s*=\s*false\s*;?)\s*\}\s*let\s*result/);
```

Moving test for the _whole_ `if`, after the one testing removing of the `continueLoop...`, would allow to make stricter test for the snipped from your comment: 


````
You should remove `if` statement with its contents.

```js
assert.match(__helpers.removeJSComments(code), /rows\.push\s*\(padRow\s*\(\s*done\s*,\s*count\s*\)\s*\)\s*;?\s*\}\s*let\s*result\s*=/);
```
````